src/CMakeLists.txt

```diff
@@ -231,11 +230,14 @@ set(GAMEENGINE_SRC
     game/ui/msgbox.cpp
 )
 
-# Glob all the header files together to add to the project for msvc/xcode.
-if(CMAKE_CONFIGURATION_TYPES)
-    file(GLOB_RECURSE GAMEENGINE_HEADERS "*.h")
-endif()
+foreach(_GAMEENGINE_SOURCE IN ITEMS ${GAMEENGINE_SRC})
+    get_filename_component(_GAMEENGINE_SOURCE_PATH "${_GAMEENGINE_SOURCE}" PATH)
+    string(REPLACE "${CMAKE_SOURCE_DIR}" "" _GAMEENGINE_GROUP_PATH "${_GAMEENGINE_SOURCE_PATH}")
```

Comment:
Instead of replacing "${CMAKE_SOURCE_DIR}" with "", replace it with "Source Files" so they all get put under the Source Files folder that they are put into currently, just in the directory structure rather than flat.